services/parser-database/parser.py

```diff
@@ -0,0 +1,218 @@
+        return f"Document(\
+                id={self.id}, \
+                jurisdiction={self.jurisdiction}, \
+                last_updated={self.last_updated}, \
+                url={self.url}, \
+                shahash={self.shahash}\
+            )"
+
+
+class Article:
+    """Class for storing articles.
+    """
+    def __init__(self, number, text):
+        self.number = number
+        self.text = text
+
+
+articles_in_memory = []
+keywords_in_memory = {}
+
+
+def get_article_by_number(art_num):
+    for item in articles_in_memory:
+        if art_num == str(item["articleNumber"]):
+            return item
+    return None
+
+
+def get_articles_that_match_keywords(keywords_list):
+    """Returns articles that match the the given keyword(s)
+
+    Args:
+        keywords_list (list): Keyword(s) to look for
+
+    Returns:
+        list: articles that match such keyword(s)
+    """
+    to_return = []
+    for keyword in keywords_list:
+        if keyword in keywords_in_memory:
+            print(keyword)
+            to_return.append(keywords_in_memory[keyword])
+    return to_return
+
+
+def count_articles(pdf_text):
+    """Identifies articles and returns a list of Article objects.
+
+    Args:
+        pdf_text (string): contains the PDF text where articles
+        are to be identified.
+
+    Returns:
+        list: article objects
+    """
+    articles = []
+    article_text = ""
+    article_count = 1
+    i = 0
+    while i < len(pdf_text) - 1:
+        if (pdf_text[i] == "artículo" or pdf_text[i] == "articulo") and (
+                pdf_text[i + 1] == str(article_count) + ".-"
+                or pdf_text[i + 1] == str(article_count) + "-"
+                or pdf_text[i + 1] == str(article_count) + "."):
+            logging.info("Article #" + str(article_count) + " recognized!")
+            articles.append(Article(article_count, article_text))
+            article_text = ""
+            article_count += 1
+            i += 1
+        else:
+            article_text += " " + pdf_text[i]
+        i += 1
+    return articles
+
+
+mty_document = {
+    "hash":
+    "afafbfbdce8c40924edae00f6ce54f0c639ce42a2" +
+    "c0fbbfa6ab82ea6925827c51",
+    "jurisdiction":
+    "Monterrey",
+    "url":
+    "http://www.guadalupe.gob.mx/wp-content/up" +
+    "loads/2019/09/Nuevo-Reglamento-Homologado-1.pdf",
+}
+document_list = []
+document_list.append(mty_document)
+
+
+def parse_all_documents():
+    """Parses all documents that are specified on the DB
+    (For obvious reasons, this won't work as is while the DB is not in use)
+    (But will still parse the hardcoded document)"""
+    for document in document_list:
+        parse(document)
+
+
+def has_file_changed(past_hash):
+    """Sees if the file is different.
+
+    Args:
+        past_hash (string): hash to compare from
+
+    Returns:
+        [boolean]: [if the file has changed or not]
+    """
+    hasher = hashlib.sha256()
+    with open("regs.pdf", "rb") as pdf_file:
```

Comment:
This probably shouldn't be hardcoded anymore. I'd expect the function to take the filename in as a parameter so this can be more general purpose.

Reply:
good point, went ahead and took the Jurisdiction attribute of the document as the file name + `.pdf`